paasta_tools/deployd/master.py

```diff
@@ -272,6 +273,8 @@ def start_watchers(self):
             )
             for watcher in watcher_classes
         ]
+        self.watcher_threads = [x for x in self.watcher_threads if x.__class__.__name__ not in disabled_watchers]  # noqa
```

Comment:
I don't like that this is changing the watcher_threads right here.

Can you instead make `watcher_threads` a property so that it returns the correct value all the time, even before running the start_watchers function?